elasticsearch/run.sh

```diff
@@ -77,6 +77,8 @@ else
     exit 1
 fi
 
+sed -i -e 's/namespace:.*/namespace: '"$NAMESPACE"'/' sgconfig/sg_config.yml
```

Comment:
heads up we assume that `$NAMESPACE` is set in the ES pod. We have tested it with `elasticsearch-operator` and the variable is always set correctly.

Should be this sed added to other places?

Reply:
I think instead of adding this directly to run.sh, you should instead add a new init script under `elasticsearch/init/` e.g. `0530-init-jaeger`
Also, since this code will run in _every_ Elasticsearch instance, you should guard against the possibility of `$NAMESPACE` not being set for some reason e.g.
```bash
if [ -n "${NAMESPACE:-}" ] ; then
 sed -i -e 's/namespace:.*/namespace: '"$NAMESPACE"'/' sgconfig/sg_config.yml
fi
```
even better would be for this code to be executed only in a jaeger Elasticsearch instance, but I'm not sure how you could determine that e.g. from the environment variables available from the downward api.

Reply:
This is where there really needs to be an `init` definition for the ES operator defined by a container and entry point.  I think short term the suggestion of an init script is the best solution.  Maybe it should only be conditionally executed when an env var is present.  Can jaeger define env vars with the operator? I dont actually know whats exposed.

Reply:
Does elasticsearch-operator expose env vars in CR? I don't think it does.

Reply:
I have moved sed into `init/0001-jaeger`. There is `0010-seed-acl` which runs `es_seed_acl` jaeger script has to run before it to so the seed applies changes.